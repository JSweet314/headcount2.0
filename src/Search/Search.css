.Header {
	position: fixed;
	top: 0;
	width: 100%;
	height: 150px;
	background-color: #309B96;
	color: #FFFFFF;
	font-family: 'Special Elite', cursive;
	font-size: 25px;
	text-align: center;
}

.Search {
	position: relative;
	width: 70%;
	margin: auto;
}

.Search h1 {
	margin-bottom: 0px
}

.Search-bar,
.Search-btn {
	height: 35px;
	padding: 10px;
	background: none;
	background-border: solid 2px #FFFFFF;
	border: solid 2px #FFFFFF;
	color: #FFFFFF;
	font-family: 'Special Elite', cursive;
	font-size: 16px;
}

.Search input::placeholder {
	color: #FFFFFF;
}

.Search-bar {
	width: 80%;
}

.Search-btn {
	position: absolute;
	right: 10%;	
}

