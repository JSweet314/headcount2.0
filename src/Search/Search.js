import React from 'react';
import './Search.css';
import PropTypes from 'prop-types';

const Search = ({searchDistrict, searchValue, clearSearch}) => {
  return (
    <div>
      <input 
        value={searchValue}
        onChange={searchDistrict}
        type="search"/>
      <button
        onClick={clearSearch}
      >CLEAR</button>
    </div>
  );
};

Search.propTypes = {
  searchDistrict: PropTypes.func.isRequired,
  searchValue: PropTypes.string.isRequired,
  clearSearch: PropTypes.func.isRequired
};

export default Search;