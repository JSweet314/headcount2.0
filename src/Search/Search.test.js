import React from 'react';
import {shallow, mount} from 'enzyme';
import Search from  './Search.js';

describe('Search', () => {
  let wrapper;
  const event = { target: { value: 'hello' } };
  const mockClearSearch = jest.fn();
  const mockSearchDistrict = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Search
        searchValue={''}
        clearSearch={mockClearSearch}
        searchDistrict={mockSearchDistrict}
      />
    );
  });

  it('should match its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should receive a search value string', () => {
    expect(wrapper.props('searchValue')).toBeDefined();
  });

  it('should invoke searchDistrict when input change occurs', () => {
    const input = wrapper.find('input');

    input.simulate('change', event);

    expect(mockSearchDistrict).toHaveBeenCalledWith(event)
  });

  it('should invoke clearSearch when the clear button is clicked', () => {
    const clearButton = wrapper.find('button');

    clearButton.simulate('click');

    expect(mockClearSearch).toHaveBeenCalled();
  });
});

