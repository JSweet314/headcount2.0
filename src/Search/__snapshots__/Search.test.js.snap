// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match its snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    clearSearch={[Function]}
    searchDistrict={[Function]}
    searchValue=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          type="search"
          value=""
/>,
        <button
          onClick={[Function]}
>
          CLEAR
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "type": "search",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "CLEAR",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "CLEAR",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            type="search"
            value=""
/>,
          <button
            onClick={[Function]}
>
            CLEAR
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "type": "search",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "CLEAR",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "CLEAR",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
